R/zchunk_LA122.gasproc_refining.R

```diff
@@ -42,100 +42,370 @@ module_energy_LA122.gasproc_refining_DISABLED <- function(command, ...) {
+
+    # Data in long format and removing "X" year
+    L1011.en_bal_EJ_R_Si_Fi_Yh%>%
+    gather(year,value,-GCAM_region_ID,-fuel,-sector) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) -> L1011.en_bal_EJ_R_Si_Fi_Yh
+
+    L121.in_EJ_R_unoil_F_Yh%>%
+      gather(year,value,-GCAM_region_ID,-fuel,-sector) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) -> L121.in_EJ_R_unoil_F_Yh
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Perform computations: Will start with refining
+
+    # Most of the technologies (in calibrated_techs.csv) have inputs from outputs based on exogenous coefficients (A22.globaltech_coef).
+    # Extracting those coefficients and interpolating them to 2010.
+    # Remove "x" year
+    A22.globaltech_coef %>%
+      semi_join(select(calibrated_techs, supplysector, subsector, technology), by = c("supplysector", "subsector", "technology")) %>%
+      left_join_error_no_match(select(calibrated_techs, supplysector, subsector, technology, minicam.energy.input, sector, fuel), by = c("supplysector", "subsector", "technology","minicam.energy.input")) %>%
+      gather(Xyear, value, -supplysector, -subsector, -technology, -minicam.energy.input, -sector, -fuel) %>%
+      mutate(Xyear = as.numeric(Xyear)) %>%
+      filter(Xyear == "1971") %>%
+      repeat_add_columns(tibble::tibble(year = HISTORICAL_YEARS)) %>%
```

Comment:
So, I called it XYear but it does not actually correspond to x1971, x1972... It was just an temp name I used (instead of using Year2 or something else). I will rename it. A22.globaltech_coef has data in 1971, 2020, and 2100, but I just need the 1971 value.